dev_guide/managing_images.adoc

```diff
@@ -0,0 +1,582 @@
+= Managing Images
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+An
+link:../architecture/core_concepts/builds_and_image_streams.html#image-streams[image
+stream] comprises one or more
+link:../architecture/core_concepts/containers_and_images.html#docker-images[Docker
+images] identified by tags. It presents a single virtual view of related images,
+similar to a Docker image repository.
+
+By watching an image stream, builds and deployments can receive notifications
+when new images are added and react by performing a build or deployment,
+respectively.
+
+There are many ways you can interact with images and set up image streams,
+depending on where the images' registries are located, any authentication
+requirements around those registries, and how you want your builds and
+deployments to behave. The following sections cover a range of these topics.
+
+[[tagging-images]]
+== Tagging Images
+
+Before working with image streams and their tags, it will help to first
+understand image tags in the context of Docker generally.
+
+Docker images can have names added to them that make it more intuitive what they
+contain. When interacting directly with images using the `docker` CLI, the
+`docker tag` command essentially adds an alias to an image that can consist of
+several parts. Those parts can include:
+
+----
+<registry_host>/<user_name>/<image_name>:<tag>
+----
+
+The `<user_name>` part in the above could also refer to a
+link:../architecture/core_concepts/projects_and_users.html#projects[project] or
+link:../architecture/core_concepts/projects_and_users.html#namespaces[namespace]
+if the image is being stored in an {product-title} or Kubernetes environment.
+
+{product-title} provides the `oc tag` command, which is similar to the `docker
+tag` command, but operates on image streams instead of directly on images.
+
+[NOTE]
+====
+See Red Hat Enterprise Linux 7's
+https://access.redhat.com/documentation/en/red-hat-enterprise-linux-atomic-host/version-7/getting-started-with-containers/#creating_docker_images[Getting
+Started with Containers] documentation for more about tagging images directly
+using the `docker` CLI.
+====
+
+[[adding-tag]]
+=== Adding Tags to Image Streams
+
+Keeping in mind that an image stream in {product-title} comprises one or more
+Docker images identified by tags, an image stream can also be configured so that
+a tag "tracks" another one.
+
+For example, to configure the *ruby* image's *latest* tag to always refer to the
+current image for the tag *2.0*:
```

Comment:
There are two basic types of tags:
- "permanent" - which is the default type, this means that after this operation we're pointing to specific image in time, even when it changes this will not reflect in the tag.
- "tracking" - specified with `--alias` flag, which actually tracks the tag.